### PR TITLE
Sign in with Apple

### PR DESCRIPTION
**Pull request by louischan-oursky**

### CODE REVIEW

File: features/332-sso/api.md [hunk 1]
```diff
@@ -80,6 +78,48 @@ skygear.auth.getLoginRedirectResult().then(user => {
...
+function linkApple(providerID: string, callbackURL: string): Promise<User>;
+```
+
+### Example
+
+```typescript
+function LoginScreen() {
+  // For a React Native app, Sign in with Apple must provide two experiences.
+  //
+  // On iOS, the developer must use loginApple which uses AuthenticationServices
```
Comment (kiootic) [inline features/332-sso/api.md]: It seems there are more cases here:
- iOS >= 13: use AS
- Android / iOS < 13: use standard OIDC flow

Is it possible to unify these three cases into single call to `loginApple`?

File: features/332-sso/sign-in-with-apple-prerequisite.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+# How to set up Sign in with Apple as a developer
```
Comment (kiootic) [inline features/332-sso/sign-in-with-apple-prerequisite.md]: I'm not sure if we should include these instructions in this repo. These instructions does not seems specific to Skygear (which can be easily found by searching).

File: features/332-sso/api.md (hunk 1 above)
Comment (louischan-oursky) [reply features/332-sso/api.md]: The provider ID is different. (`apple-app` vs `apple-web`). I expect the developer to use [react-native-device-info](https://github.com/react-native-community/react-native-device-info) to check the version themselves.

File: features/332-sso/sign-in-with-apple-prerequisite.md (hunk 2 above)
Comment (louischan-oursky) [reply features/332-sso/sign-in-with-apple-prerequisite.md]: All contents are searchable, but scattered in different places. The skygear specific instruction is the UserConfiguration example.

File: features/332-sso/api.md (hunk 1 above)
Comment (kiootic) [reply features/332-sso/api.md]: Is there any reason that the provider IDs have to be different? Would there be problems if developer would want to link the principals of the same Apple ID signed up using `apple-app` & `apple-web` together?

Comment (louischan-oursky) [reply features/332-sso/api.md]: The opaque user ID we obtain from Apple is scoped under the team. So as long as the team remains the same, different App IDs and different Services IDs still get the same opaque user ID.

Comment (louischan-oursky) [reply features/332-sso/api.md]: I have updated the example to use react-native-device-info

**Pull request merged**